source/code/providers/support/mycnf-scanner.cpp

```diff
@@ -0,0 +1,244 @@
+
+                        // Starting new section - set defaults
+                        portNum = 3306;
+                        bindAddress = L"127.0.0.1";
+                    }
+                    continue;
+                }
+
+                std::vector<std::wstring> elements;
+                SCXCoreLib::StrTokenize(line, elements, L"=");
+                if ( elements.size() != 2 )
+                {
+                    // Not a line that we care about
+                    continue;
+                }
+
+                // Check for a port declaration (port = xxx)
+                if ( 0 == SCXCoreLib::StrCompare(L"port", elements[0], true) )
+                {
+                    SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"   Found Port: ", elements[1]));
+                    portNum = SCXCoreLib::StrToUInt( elements[1] );
+                }
+
+                // Check for a bind-address declaration (bind-address = 172.0.0.1)
+                if ( 0 == SCXCoreLib::StrCompare(L"bind-address", elements[0], true) )
+                {
+                    SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"   Found bind-address: ", elements[1]));
+                    bindAddress = elements[1];
+                }
+            }
+
+            if ( portNum )
+            {
+                SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(SCXCoreLib::StrAppend(L"   Flushing port ", portNum).append(L", Bind Address: "), bindAddress));
+
+                auth.AddCredentialSet(portNum, SCXCoreLib::StrToUTF8(bindAddress));
+                ports[portNum] = bindAddress;
+            }
+        }
+
+        return true;
+    }
+
+    void MySQL_ConfigurationScanner:: LoadCnfDirectory(
+        const SCXCoreLib::SCXFilePath& directory, 
+        MySQL_Authentication& auth,
+        std::set< std::wstring >& visited, 
+        std::map<unsigned int, std::wstring>& ports)
+    {
+        try
+        {
+            SCX_LOGHYSTERICAL(m_log, SCXCoreLib::StrAppend(L"LoadCnfFilesFromDirectory", directory.Get()));
+            std::vector< SCXCoreLib::SCXFilePath > cnfFiles = m_extDepends->GetFiles(directory);
+
+            for( std::vector< SCXCoreLib::SCXFilePath >::const_iterator iter = cnfFiles.begin(); iter != cnfFiles.end(); ++iter)
+            {
+                std::wstring fileSuffix = iter->GetFilesuffix();
+                if (fileSuffix == L"cnf")
+                {
+                    LoadCnfFile(*iter, auth, visited, ports);
+                }
+                else
+                {
+                    SCX_LOGHYSTERICAL(m_log,SCXCoreLib::StrAppend(iter->Get(), L" is not a cnf file"));
+                }
+            }
+        }
+        catch (SCXCoreLib::SCXException & e)
+        {
+            std::wstringstream ss;
+            ss << L"Encountered " << e.What() << L" when trying to read entried from " << directory.Get();
+            SCX_LOGERROR(m_log, ss.str());
```

Comment:
As I mentioned above,  I was deliberately doing that.  MySQL documentation does not say what it will do if ,say, the directory does not present. If it ignores such errors, we will be wrong in throwing exception and bailing out. In general, we must be more lenient in reading the cnf than MySQL. 
Anyways, I will change it.  The problem here is the options file is under-specified. I will send out a new iteration.